pinot-segment-local/src/test/java/org/apache/pinot/segment/local/segment/index/loader/SegmentPreProcessorTest.java

```diff
@@ -304,6 +306,28 @@ public void testEnableFSTIndexOnExistingColumnDictEncoded()
     checkFSTIndexCreation(EXISTING_STRING_COL_DICT, 9, 4, _newColumnsSchemaWithFST, false, false, 26);
   }
 
+  @Test
+  public void testForwardIndexHandler()
+      throws Exception {
+    Map<String, ChunkCompressionType> compressionConfigs = new HashMap<>();
+    ChunkCompressionType newCompressionType = ChunkCompressionType.ZSTANDARD;
+    compressionConfigs.put(EXISTING_STRING_COL_RAW, newCompressionType);
+    _indexLoadingConfig.setCompressionConfigs(compressionConfigs);
+    _indexLoadingConfig.setNoDictionaryColumns(new HashSet<String>() {{
+      add(EXISTING_STRING_COL_RAW);
+    }});
+
+    // Test1: Rewriting forward index will be a no-op for v1 segments. Default LZ4 compressionType will be retained.
+    constructV1Segment();
+    checkForwardIndexCreation(EXISTING_STRING_COL_RAW, 5, 3, _schema, false, false, false, 0, ChunkCompressionType.LZ4);
+
+    // Convert the segment to V3.
+    new SegmentV1V2ToV3FormatConverter().convert(_indexDir);
+
+    // Test2: Now forward index will be rewritten with ZSTANDARD compressionType.
```

Comment:
Can we also cover the following scenarios if not already in `ForwardIndexHandler` ?

- The column on which a `CHANGE_RAW_COMPRESSION_TYPE` action is needed, another orthogonal action on the same column is needed (e.g range index or a text index which can be enabled on noDict columns) by the default column handler. Run some queries that touch the column in SELECT and WHERE to leverage the changed fwd index (with new comp codec) and newly added inv index. Basically just to ensure that newly introduced handler works with the existing handlers and nothing is broken, their sequencing is correct etc. 

- The column on which a `CHANGE_RAW_COMPRESSION_TYPE` action is needed, another orthogonal action on a different column is needed. This is again to ensure we haven't somehow messed with existing handlers across same or different columns.

Reply:
Can we also test that there is no `inprogress` file and `fwd-index` is not in v1 form ?

Reply:
Added these tests. We verify various metadata, etc after segmentPreprocessor.process(). This verifies if the existing handlers work with ForwardIndexHandler. 
However, the test doesn't issue queries to validate. If you feel that's needed, I can add a brand new test in OfflineClusterIntegrationTest to wire querying and checking.

Reply:
So one concern (and please clarify if this is a non-issue) I had was for the scenario I asked for in first point

Once the ForwardIndexHandler finishes rewriting fwd index with new comp codec and we return
- the new fwd index is in v3 format
- the entry for old fwd index is removed from in-memory map
- the index_map file has 2 entries for fwd index on that column (old and new)

Next handler (RangeIndexHandler in this case) gets invoked and reads the forward index to generate range index. I am guessing this will read the new forward index (because old one is no longer accessible via the _columnEntries map). 

Basically, my concern was if the range index gets generated incorrectly / corrupt because of 2 forward indexes (since sweep hasn't been done as of this time). But looks like this can't happen

Reply:
> I am guessing this will read the new forward index (because old one is no longer accessible via the _columnEntries map).
That is right. Subsequent handlers (like range index, inv index, etc) will only see the newly created forward because all lookups are done using the in-memory `_columnEntries` in `SingleFileIndexDirectory.java`

Given that this test verifies text index creation on the same column where compressionType has been changed, we should be good here.